client/components/SendZoopDialog.tsx

```diff
@@ -0,0 +1,101 @@
+import SendIcon from "@mui/icons-material/Send"
+import Button from "@mui/material/Button";
+import Dialog from "@mui/material/Dialog";
+import DialogActions from "@mui/material/DialogActions";
+import DialogContent from "@mui/material/DialogContent";
+import DialogTitle from "@mui/material/DialogTitle";
+import TextField from "@mui/material/TextField";
+import React, { useState } from "react";
+
+import { 
+    useGetAllUsersQuery, 
+    useGetMeQuery, 
+    useSendZoopMutation 
+} from "../features/api";
+import { Zoop } from "../../src/types/custom";
+
+export interface SendZoopDialogProps {
+    openDialog: boolean
+    isZoopSent: () => void
+    sentZoop: (zoop: Zoop) => void;
+}
+
+const SendZoopDialog = ({openDialog, isZoopSent, sentZoop}: SendZoopDialogProps) => {
+    const [username, setUsername] = useState("");
+    const [content, setContent] = useState("");
+    const [open, setOpen] = useState(openDialog);
+
+    const currentUser = useGetMeQuery().data;
+
+    const [sendZoop, { isLoading: isSendZoopLoading, isError, data: zoopData }] = useSendZoopMutation();
+    const {data: usersData, isLoading: isGetAllUsersLoading, error } = useGetAllUsersQuery(); 
+
+    const handleSendZoopClick = () => {
+        setOpen(false);
+        if (!isGetAllUsersLoading && !usersData) {
+            throw new Error('No users to send Zoop to');
+        } else if (!isGetAllUsersLoading && usersData) {
+            const [recipient] = usersData.users.filter(el => el.user.username === username);
+            if (recipient && currentUser) {
+                sendZoop({
+                    content: content,
+                    authorId: currentUser.user.id,
+                    receiverId: recipient.user.id
+                })
+                isZoopSent();
+                // TODO: figure out why zoopData is undefined.
+                if (!isSendZoopLoading && zoopData) {
+                    console.log("zoopData: ", zoopData);
+                    sentZoop(zoopData.zoop);
+                }
+            } else {
+                throw new Error('No such user exists. Please select a different username');
+            }  
+        }  
+    }
+
+    return (
+        <Dialog
+            open={open}
+            onClose={() => setOpen(false)}
```

Comment:
You should also call the `onClose` from the parent - this will avoid this component remaining mounted (even if it's not visible) after it was closed